docs/rfcs/0002-session-id-sharing.md

```diff
@@ -0,0 +1,223 @@
+to the actual session ping.
+
+#### On session end
+
+Once an ongoing session is ended,
+
+1. The `daemonsession` ping is submitted with the reason `daemon_end`.
+2. The daemon's local storage for the `shared_session_id` is cleared.
+3. The `shared_session_id` metric is set to a known value. Due to its `User` lifetime,
+   this metric cannot be cleared from storage ever. To indicate a period of inactivity,
+   we will leverage a known UUID.
+
+#### On daemon process start
+
+The session end trigger may not be hit, if the daemon process is force killed. Therefore,
+whenever the daemon process is started the local storage will be searched for a lingering
+`shared_session_id` value. That is a clear signal that the session end trigger was not hit
+and that there may be lingering data from a previous session on the Glean storage.
+
+When that is the case,
+
+1. The lingering `shared_session_id` value is recorded to Glean.
+2. The `daemonsession` ping is submitted with the reason `flush_init`.
+
+### On the main application
+
+The same three triggers which are important for the daemon and the `daemonsession` ping,
+are also important for the main application and the `vpnsession` ping.
+
+#### On session start
+
+If the main application is active during a session start it will,
+
+1. Signal to the daemon that a new session is to be started.
+2. It will then ask the daemon for the value of the shared session id it created.
+3. If the previous two communications where successful
+   1. The `vpnsession` ping will be submitted with reason `flush`.
+   2. The value of the retrieved `shared_session_id` is recorded.
+   3. Other session metrics may be recorded from this point.
+   4. If the communication with the daemon fails, an error metric is recorded.
+
+> **Note**: If the retrieved session id is the known inactive session id,
+> this is considered an unsuccessful retrieval and the error path is taken.
+
+#### On session end
+
+If the main application is active during a session end it will,
+
+1. Submit the session ping with reason `end`.
+2. Set `shared_session_id` metric is set to a known value. The same known value used for inactive
+   sessions on the `daemonsession` ping.
+
+#### On application start
+
+When the main application is started, just like in the daemon, there may be some lingering data
+from a previous session in storage. On the main application, the session_id is not directly managed,
+so there is no way to know if there is lingering data or not.
+
+In this case, the main application will submit the `vpnsession` ping with reason `flush_init` on
+_every_ initialization. In case there is no lingering data, this is not really an issue. The ping
+will be sent with only the currently stored session id value.
+
+After sending the `flush_init` ping, the VPN will query the daemon for a session id.
+If a session id is received, that means there is an ongoing VPN session. In that case, the
```

Comment:
Won't a session ID always be received? It's just whether it's the known inactive session value or not.

Reply:
Correct. But the daemon may not be running. In that case, no session id is received.